server/controllers/officecontroller.js

```diff
@@ -18,24 +18,24 @@ class OfficeController {
     const {
       type, name,
     } = req.body;
-
-    const id = officeDb.length + 1;
-    const officeData = {
-      id,
+    const newOffice = OfficeHelper.createOffice({
       type,
       name,
-    };
-
-    officeDb.push(officeData);
-
+    }); if (!req.body.type) {
```

Comment:
Create another folder called middleware that will contain file that will do the validation before getting to the controller. Import the middleware file in the route example `router.post('/parties', partyMiddleware.validateCreateParty, PartyController.postParty)` . So the req will pass through the middleware validation before getting to the controller. Remember to call `next()` function if all data is valid. That will pipe the request to the controller

Reply:
Okay. Can I merge this branch then create another branch to do the validation middleware thing?